### ISSUE TITLE
Collapsible FAQ, and ToC with nav links

### ISSUE THREAD

**Issue by jaseiler** [researchcompendia/researchcompendia#54]
The FAQ needs to have a ToC with id navigation, there is already too much there, and the will be many more to come.  
Having the questions collapsible is optional too, but a Table of Contents is a must.

**Comment by jaseiler** [comment]
Also, manual ordering would be nice, they currently appear in order of oldest to newest edited

**Comment by codersquid** [comment]
I can put the collapsible templating back in.

**Comment by codersquid** [comment]
Definitely agree on the manual ordering.

**Comment by codersquid** [comment]
I had removed the accordion functionality because I wasn't sure it was necessary for the functionality of making the page not being a big wall of text. versus something that looked flashy but got in the way. a TOC would help with a page that seemed to be a big wall of text.

I'm including all the following detail in case this sounds interesting to pick up, since it can be done independently while I'm working on other things.

Also, I will be at a hackathon tomorrow, and there is an open hack session. Perhaps I can introduce a friend to the project and see if they'd like to work on the FAQ there with  me to review.

Here's the current FAQ model. We could add an index field then specify ordered-by on the index field.

```
class Faq(TimeStampedModel):
    # inherits created and modified DateTimeFields fields from TimeStampedModel
    question = models.TextField(max_length=200)
    answer = models.TextField(
        max_length=3000,
        help_text=_(u'The answer to the question can be written in Markdown syntax'))

    def __unicode__(self):
        return self.question[:10]
```

Here's the old template code for the faq, we'd remove the classes that specify green. Also, the fontawesome classes need to be updated according to the most recent fontawesome classes. that's not difficult, I have it done in other places.

```
{% extends "base.html" %}
{% load home_tags %}

{% block title %}FAQ{% endblock %}

{% block content %}

<h1>FAQ</h1>
<div class="panel-group" id="accordion">
{% for faq in faq_list %}
<div class="panel">
    <div class="panel-heading">
      <h4 class="panel-title">
        <a class="accordion-toggle" data-toggle="collapse" data-parent="#accordion" href="#{{ forloop.counter }}">
            {% if forloop.first %}
            <div class="icon-plus-sign-alt rmc-green-font" id="iconstate"> {{ faq.question }}</div>
            {% else %}
            <div class="icon-minus-sign-alt rmc-green-font" id="iconstate"> {{ faq.question }}</div>
            {% endif %}
        </a>
      </h4>
    </div> <!-- /panel-heading -->
    {% if forloop.first %}
    <div id="{{ forloop.counter }}" class="panel-collapse collapse in">
    {% else %}
    <div id="{{ forloop.counter }}" class="panel-collapse collapse">
    {% endif %}
      <div class="panel-body">
        {{ faq.answer|md }}
      </div> <!-- /panel-body -->
    </div> <!-- /panel-collapse -->
  </div> <!--/panel-->
{% empty %}
<p>There are no FAQs</p>
{% endfor %}
</div> <!-- /panel-group -->
{% endblock content %}
```

**Comment by codersquid** [comment]
@jaseiler I think you mentioned having nav/toc links on the right in another issue, and I'd like to know what your wishes are for it with regards to the FAQ.

The bootstrap [docs navigation](http://getbootstrap.com/components/) uses a right toc with collapsible elements for sub topics. Their toc is also fixed, and uses  [scrollspy](http://getbootstrap.com/javascript/#scrollspy).

Let me know what you think of scrollspy along with the collapsible feature. I like it, but one problem I have with fixed collapsible nav with scrollspy happens when I want to go back to a page to directly jump to a topic I know is there, but which is towards the bottom of the nav. I hate that part of the experience so much that I'm not sure I want to inflict it on our users. But since I also like other parts of the experience, I would like to know your thoughts about what you'd like to see implemented and also whether you think the problem I have is even a problem for our content.

To try this experience out, pretend like you want to find the docs on alerts but don't have that [#alerts](http://getbootstrap.com/components/#alerts) link to jump to. Alerts is towards the bottom of a long toc nav, and I can't go to that section of the screen to scroll down to it. Based on that, I end up scrolling the center content for a while until I hit the alerts section.

I like the scrollspy experience for tocs that are short and fit in the window, but not for tocs that are long and don't fit in the window. If we have a lot of questions, that will happen to us. Do you want to have scrollspy but maybe without expanding sections -- instead have those in side navs that can open optionally? And then we could organize questions in to categories, which could be what we affix to the side with the highlighting bit? Or, don't use the idea of a fixed floating toc, but keep other aspects of the experience?

Let me know if that is something that appeals. I've experienced some of those things on other sites as well.

**Comment by jaseiler** [comment]
I like scrollspy a lot.  I think having the ToC is more important than the collapsibility, and if I'm understanding your issue correctly, not having the collapsibility fixes the problem, right?  Because than it always jumps to the right place.
I'd be happy with scrollspy nav and non-collapsing issues.

**Comment by codersquid** [comment]
Ready to review. code here https://github.com/researchcompendia/tyler/commit/de3ff997c061cef49e6e52f7e191a58f783a13d9 and website here http://preprod-compendia.herokuapp.com/faq/

**Comment by jaseiler** [comment]
1) It would be nice if all the questions in the ToC fit on the visible page for your average screen.  So maybe a smaller font and/or smaller line spacing.

2) Navigates too low on the page so that the question is hidden under the top bar.  I've always hacked solutions to this issue by placing the ids on shifted up spans or divs, but is there a better solution?

3) What about the manual sorting of questions through the admin page issue?  Is that tough to resolve.

**Comment by codersquid** [comment]
The manual sorting is clunky, but you can do it in the admin by saving the index number by hand. That is extremely clunky.

**Comment by codersquid** [comment]
I don't have scrollspy and collapsability working yet, so this commit mainly serves to have the toc as sidenav

**Comment by codersquid** [comment]
It looks like the obscuring of the question happens due to the fixed top navbar.
- What are your thoughts on getting rid of the fixed top navbar?

**Comment by codersquid** [comment]
For now so that our production site has a nicer looking faq page that it currently does, I'm going to checkin a change with all collapsed questions (so that it matches your request for every question appearing on the first look on an average big screen) and when the toc right-nav is floating and using scrollspy with the collapse/expand feature we can push that out. OR, if you like the toc right-nav without collapse/expand and we can do that sooner.

**Comment by codersquid** [comment]
I've got this deployed to researchcompendia.org now.

(oh man, you know how everyone says don't deploy something before a holiday? it's true. I deployed the code, and forgot to migrate the database. I migrated the database and didn't realize I had the wrong database in my environment variable. etc. oh dear. I'm glad I got that all straightened out. That stuff never seems to happen so much when _not_ deploying right before a holiday.)
